### ISSUE TITLE
make failing with Undeclared constants error on CentOS

### ISSUE THREAD

**Issue by tusharmarda** [lz4/lz4#1092]
Getting following error with `make` command

```
make[1]: Entering directory `/usr/local/scaligent/toolchain/lz4/wrkdir/lz4-repo/lib'
make[1]: Nothing to be done for `lib-release'.
make[1]: Leaving directory `/usr/local/scaligent/toolchain/lz4/wrkdir/lz4-repo/lib'
make[1]: Entering directory `/usr/local/scaligent/toolchain/lz4/wrkdir/lz4-repo/programs'
/usr/local/scaligent/toolchain/crosstool/v4/x86_64-unknown-linux-gnu/bin/x86_64-unknown-linux-gnu-gcc -O3   -D_GLIBCXX_USE_CXX11_ABI=0 -I/usr/local/scaligent/toolchain/local/include -fno-omit-frame-pointer -I../lib -DXXH_NAMESPACE=LZ4_  -c -o lz4io.o lz4io.c
lz4io.c: In function 'LZ4IO_skipBlocksData':
lz4io.c:1398:29: error: 'LZ4F_BLOCK_HEADER_SIZE' undeclared (first use in this function)
     unsigned char blockInfo[LZ4F_BLOCK_HEADER_SIZE];
                             ^
lz4io.c:1398:29: note: each undeclared identifier is reported only once for each function it appears in
lz4io.c:1407:83: error: 'LZ4F_BLOCK_CHECKSUM_SIZE' undeclared (first use in this function)
             const unsigned long nextBlock = nextCBlockSize + (blockChecksumFlag * LZ4F_BLOCK_CHECKSUM_SIZE);
                                                                                   ^
lz4io.c:1412:44: error: 'LZ4F_CONTENT_CHECKSUM_SIZE' undeclared (first use in this function)
                     if (UTIL_fseek(finput, LZ4F_CONTENT_CHECKSUM_SIZE, SEEK_CUR) != 0) {
                                            ^
make[1]: *** [lz4io.o] Error 1
make[1]: Leaving directory `/usr/local/scaligent/toolchain/lz4/wrkdir/lz4-repo/programs'
make: *** [lz4-release] Error 2
```

**Comment by Cyan4973** [comment]
This is strange.
There shouldn't be any relation between `centos` as a target and the kind of error you are reporting.
And fyi, I've been unable to reproduce this compilation error on any of my test systems.

These constants are defined in `lz4frame.h`.
`lz4frame.h` is properly included within `lz4io.c`.
So it's that straightforward, there is simply no room for platform-specific behavior.

**Comment by tusharmarda** [comment]
I had included `centos` since I had seen some other issue mentioning the OS they were building on.

Regarding my issue, my system already had v1.9.1 installed when I tried to install the latest. Once I cleaned+uninstalled v1.9.1, it ran fine.

Your points are right, I realised the same thing after posting the issue. So I tried to install manually using the make+make install commands and it worked. But it failed with the install script. I still do not understand fully why an existing previous version would cause this error, but since the standard commands work for a clean install, I'm closing this issue.

Thanks for the prompt response :)
